samples/benchmarks/coremark/README.rst

```diff
@@ -0,0 +1,146 @@
+
+CoreMark has the following predefined run types and data sizes that are used for data algorithms:
+
+* The :kconfig:option:`CONFIG_COREMARK_RUN_TYPE_PERFORMANCE` Kconfig option - Sets :kconfig:option:`CONFIG_COREMARK_DATA_SIZE` to 2000 bytes.
+* The :kconfig:option:`CONFIG_COREMARK_RUN_TYPE_PROFILE` Kconfig option - Sets :kconfig:option:`CONFIG_COREMARK_DATA_SIZE` to 1200 bytes.
+* The :kconfig:option:`CONFIG_COREMARK_RUN_TYPE_VALIDATION` Kconfig option - Sets :kconfig:option:`CONFIG_COREMARK_DATA_SIZE` to 500 bytes.
+
+You can also specify a custom :kconfig:option:`CONFIG_COREMARK_DATA_SIZE` value and submit your results by following the rules from the `CoreMark GitHub`_ repository.
+Make sure that when setting the :kconfig:option:`CONFIG_COREMARK_DATA_SIZE` Kconfig option the required memory space is available.
+
+You can select the following memory allocation methods:
+
+* The :kconfig:option:`CONFIG_COREMARK_MEMORY_METHOD_STACK` Kconfig option uses the main thread stack for data allocation and is enabled by default.
+  You can change the main stack size by setting the :kconfig:option:`CONFIG_MAIN_STACK_SIZE` Kconfig option.
+* The :kconfig:option:`CONFIG_COREMARK_MEMORY_METHOD_MALLOC` Kconfig option uses Zephyr :c:func:`k_malloc\k_free` functions to allocate the memory in the heap.
+  You can adjust the heap size with the :kconfig:option:`CONFIG_HEAP_MEM_POOL_SIZE` Kconfig option.
+* The :kconfig:option:`CONFIG_COREMARK_MEMORY_METHOD_STATIC` option allows data allocation in the RAM as a regular static variable.
+  In this case you do not need to be aware of the main thread stack size or heap memory pool size.
+
+CoreMark can also be executed in the multithread mode.
+To specify a number of threads, use the :kconfig:option:`CONFIG_COREMARK_THREADS_NUMBER` Kconfig option.
+You can configure up to four threads.
+In the multithread mode, CoreMark is executed in all threads simultaneously.
+By default, the :kconfig:option:`CONFIG_TIMESLICE_SIZE` Kconfig option is set to 10 ms which imitates RTOS multithread usage.
+However, in the final report the thread execution is combined.
+In the multithread mode, the :kconfig:option:`CONFIG_COREMARK_DATA_SIZE` Kconfig option is allocated for each thread separately.
+Only the :kconfig:option:`CONFIG_COREMARK_MEMORY_METHOD_STACK` and :kconfig:option:`CONFIG_COREMARK_MEMORY_METHOD_MALLOC` memory methods can be used with multiple threads.
+In case of the :kconfig:option:`CONFIG_COREMARK_MEMORY_METHOD_STACK` Kconfig option, the data for all threads are allocated to the main thread stack.
+
+Building and running
+********************
+
+When running the benchmark, an extra build flag (:kconfig:option:`CONFIG_COMPILER_OPT`) is set to ``-O3`` to achieve the best CoreMark results.
+
+.. code-block:: sh
+
+   west build -b nrf5340dk_nrf5340_cpuapp samples/benchmark/coremark
+   west flash
+
+After flashing, messages describing the benchmark state will appear in the console.
```

Comment:
Do not duplicate generic information about building and flashing. Please link to https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/config_and_build/configuring_app/index.html and https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/config_and_build/programming.html instead.

Reply:
Also, please run the sample using the nRF Connect extension for VS Code and make sure no additional configuration steps are required in the extension.

Reply:
Should I use approach as in other samples here in hello_world:

.. |sample path| replace:: :file:`samples/multicore/hello_world`

.. include:: /includes/build_and_run_sb.txt